const userService = require('../services/User.service');

const getById = async (req, res) => {
  try {
    const { id } = req.params;

    const user = await userService.getById(id);
  
    if (!user) {
      return res.status(404).json({ message: 'User not found' });
    }
  
    return res.status(200).json(user);
  } catch (error) {
    console.log(error.message);
    return res.status(500).json({ message: 'Internal error' });
  }
};

const createUser = async (req, res) => {
  try {
    const { name, email, password, address } = req.body;
    
    const user = await userService.createUser(name, email, password, address);

    if (!user) {
      return res.status(500).json({ message: 'Erro no service createUser' });
    }

    return res.status(201).json({ message: `UsuÃ¡rio ${user.name} criado com sucesso.` });
  } catch (error) {
    console.log(error.message);
    return res.status(500).json({ message: 'Internal error' });
  }
};

const login = async (req, res) => {
  try {
    const { email, password } = req.body;

    const userToken = await userService.login(email, password);

    if (!userToken) {
      return res.status(401).json({ message: 'Incorrect username or password' });
    }

    return res.status(200).json(userToken);
  } catch (error) {
    console.log(error.message);
    return res.status(500).json({ message: 'Internal error' });
  }
}

module.exports = {
  getById,
  createUser,
  login,
}